sqlglot/dialects/starrocks.py

```diff
@@ -13,11 +13,58 @@
 )
 from sqlglot.dialects.mysql import MySQL
 from sqlglot.helper import seq_get
+from sqlglot.tokens import TokenType
+
+
+def _duplicate_key_sql(self, expression: exp.DuplicateKeyProperty) -> str:
+    expressions = self.expressions(expression, flat=True)
+    options = self.expressions(expression, key="options", flat=True, sep=" ")
+    options = f" {options}" if options else ""
+    return f"DUPLICATE KEY ({expressions}){options}"
+
+
+def _distributed_by_hash_sql(self, expression: exp.DistributedByHashProperty) -> str:
+    expressions = self.expressions(expression, key="expressions", flat=True)
+    sorted_by = self.expressions(expression, key="sorted_by", flat=True)
+    sorted_by = f" ORDER BY ({sorted_by})" if sorted_by else ""
+    buckets = self.sql(expression, "buckets")
+    # Since StarRocks v2.5.7, the number of buckets is AUTO by default.
+    # https://docs.starrocks.io/docs/sql-reference/sql-statements/data-definition/CREATE_TABLE/#distribution_desc
+    if expression.auto_bucket:
+        buckets = "AUTO"
+    # [DISTRIBUTED BY HASH (k1[,k2 ...]) [BUCKETS num] [ORDER BY (k1[,k2 ...])]]
+    return f"DISTRIBUTED BY HASH ({expressions}) BUCKETS {buckets}{sorted_by}"
+
+
+def _distributed_by_random_sql(self, expression: exp.DistributedByRandomProperty) -> str:
+    expressions = self.expressions(expression, flat=True)
+    sorted_by = self.expressions(expression, key="sorted_by", flat=True)
+    sorted_by = f" ORDER BY ({sorted_by})" if sorted_by else ""
+    buckets = self.sql(expression, "buckets")
+    if expression.auto_bucket:
+        buckets = "AUTO"
+    # [DISTRIBUTED BY RANDOM (k1[,k2 ...]) [BUCKETS num] [ORDER BY (k1[,k2 ...])]]
+    return f"DISTRIBUTED BY RANDOM ({expressions}) BUCKETS {buckets}{sorted_by}"
+
+
+def _property_sql(self: MySQL.Generator, expression: exp.Property) -> str:
+    # [PROPERTIES (['key1'='value1', 'key2'='value2', ...])]
+    return f"{self.property_name(expression, string_key=True)}={self.sql(expression, 'value')}"
```

Comment:
What is this achieving? I'm not sure if it's correct, `Property` is a pretty generic class to override.

Reply:
Yes. This case is used to set the parameter string_key to True.
I refer to the code from `sqlglot.dialects.hive._property_sql`.

Reply:
I see, I'll change that.